src/components/CalendarEvent.js

```diff
@@ -24,17 +24,21 @@ class CalendarEvent extends React.Component {
     const sharedArguments = [null, this, this.props.eventData, this.props.day];
 
     return (
-            <div className={eventClasses}
-                 onClick={this.props.onClick.bind(...sharedArguments)}
-                 onMouseOut={this.props.onMouseOut.bind(...sharedArguments)}
-                 onMouseOver={this.props.onMouseOver.bind(...sharedArguments)}
-            >
-                <div className="event-title">
-                    {title}    
-                </div>
+      <div className={eventClasses}
+        onClick={e => {
+          const [ , ...args ] = sharedArguments;
```

Comment:
Although totally valid this destructing syntax seems a little jarring.  We techincally don't need it.  You should just be able to use call with sharedArguments.  

```this.props.onClick.call(...sharedArguments);```